packages/flutter/lib/src/services/message_codecs.dart

```diff
@@ -0,0 +1,421 @@
+  //   encoding of the list elements themselves, end-to-end with no additional
+  //   type information, using big-endian two's complement or IEEE 754 as
+  //   applicable.
+  // * Lists are encoded by first encoding their length in the expanding format,
+  //   then follows the recursive encoding of each element value, including the
+  //   type byte (Lists are assumed to be heterogeneous).
+  // * Maps are encoded by first encoding their length in the expanding format,
+  //   then follows the recursive encoding of each key/value pair, including the
+  //   type byte for both (Maps are assumed to be heterogeneous).
+  static const int _kNull = 0;
+  static const int _kTrue = 1;
+  static const int _kFalse = 2;
+  static const int _kInt32 = 3;
+  static const int _kInt64 = 4;
+  static const int _kLargeInt = 5;
+  static const int _kFloat64 = 6;
+  static const int _kString = 7;
+  static const int _kUint8List = 8;
+  static const int _kInt32List = 9;
+  static const int _kInt64List = 10;
+  static const int _kFloat64List = 11;
+  static const int _kList = 12;
+  static const int _kMap = 13;
+
+  const StandardMessageCodec();
+
+  @override
+  ByteData encodeMessage(dynamic message) {
+    if (message == null)
+      return null;
+    final WriteBuffer buffer = new WriteBuffer();
+    _writeValue(buffer, message);
+    return buffer.done();
+  }
+
+  @override
+  dynamic decodeMessage(ByteData message) {
+    if (message == null)
+      return null;
+    final ReadBuffer buffer = new ReadBuffer(message);
+    final dynamic result = _readValue(buffer);
+    if (buffer.hasRemaining)
+      throw new FormatException('Message corrupted');
+    return result;
+  }
+
+  static void _writeSize(WriteBuffer buffer, int value) {
+    assert(0 <= value && value < 0xffffffff);
+    if (value < 254) {
+      buffer.putUint8(value);
+    } else if (value < 0xffff) {
+      buffer.putUint8(254);
+      buffer.putUint8(value >> 8);
+      buffer.putUint8(value & 0xff);
+    } else {
+      buffer.putUint8(255);
+      buffer.putUint8(value >> 24);
+      buffer.putUint8((value >> 16) & 0xff);
+      buffer.putUint8((value >> 8) & 0xff);
+      buffer.putUint8(value & 0xff);
+    }
+  }
+
+  static void _writeValue(WriteBuffer buffer, dynamic value) {
+    if (value == null) {
+      buffer.putUint8(_kNull);
+    } else if (value is bool) {
+      buffer.putUint8(value ? _kTrue : _kFalse);
+    } else if (value is int) {
+      if (-0x7fffffff <= value && value < 0x7fffffff) {
+        buffer.putUint8(_kInt32);
+        buffer.putInt32(value);
+      }
+      else if (-0x7fffffffffffffff <= value && value < 0x7fffffffffffffff) {
+        buffer.putUint8(_kInt64);
+        buffer.putInt64(value);
+      }
+      else {
+        buffer.putUint8(_kLargeInt);
+        final List<int> hex = UTF8.encoder.convert(value.toRadixString(16));
+        _writeSize(buffer, hex.length);
+        buffer.putUint8List(hex);
```

Comment:
I agree. It's a temporary solution. Make it work, make it right, make it fast...